eng/tools/rush-runner/src/helpers.js

```diff
@@ -60,33 +62,51 @@ const restrictedToPackages = [
 ];
 
 /**
- * Helper function that determines the rush command flag to use based on each individual package name for the 'build' check.
+ * Helper function that determines the rush command flag to use based on each individual package name
  *
  * If the targeted package is one of the restricted packages with a ton of dependents, we only want to run that package
  * and not all of its dependents.
  * @param {string[]} packageNames - An array of strings containing the packages names to run the action on.
- * @param {string[]} actionComponents - An array of strings containing the packages names to run the action on.
+ * @param {string} action - The action being performed ("build", "build:test", "build:samples", "unit-test:node", "unit-test:browser"
+ * @param {string[]} serviceDirs - An array of strings containing the serviceDirs affected
  */
-export const getDirectionMappedPackages = (packageNames, actionComponents) => {
+export const getDirectionMappedPackages = (packageNames, action, serviceDirs) => {
```

Comment:
I combined the handling for build and test here since the existing unit tests implied they should be handled this way, not sure if it's the best but it seems to work ok for now